src/converse-chatview.js

```diff
@@ -855,6 +856,20 @@ converse.plugins.add('converse-chatview', {
                 }
             },
 
+            async onMuteUser (message) {
+                try {
+                    await this.model.onMuteUser(message);
+                    const jid = message.get('from');
+                    const { messages } = this.model;
+                    messages.each(msg => {
+                        const from = msg.get('from');
+                        if (from === jid) msg.destroy();
+                    });
```

Comment:
Please keep only the `await` line inside the `try` block. Generally you want to keep your try/catch blocks as small as possible to prevent catching and swallowing errors that should normally bubble up.

Also, the message deletion code should also go into the headless part.